### PR TITLE
Index Service Agents Handler 159751456

### PR DESCRIPTION
**Pull request by chrisgilmerproj**
## Description

Add a handler to get all service agents for a shipment belonging to a TSP.

## Reviewer Notes

The test case only counts the number of returned values.  It doesn't check the contents because those are auto-generated.  Not sure how I'd do a more comprehensive test at this stage.

## Setup

You must first generate data:

```sh
make db_dev_reset db_dev_migrate && ./bin/generate-test-data -scenario=7
```

Then find a shipment that will work with the query based on the TSP user's email.

```sql
SELECT 
  service_agents.id AS service_agent, 
  shipments.id AS shipment, 
  shipment_offers.id AS shipment_offer, 
  transportation_service_providers.id AS tsp_id, 
  tsp_users.email AS tsp_email 
FROM service_agents 
LEFT JOIN shipments ON service_agents.shipment_id=shipments.id 
LEFT JOIN shipment_offers ON shipment_offers.shipment_id = shipments.id 
LEFT JOIN transportation_service_providers ON transportation_service_providers.id = shipment_offers.transportation_service_provider_id 
LEFT JOIN tsp_users ON tsp_users.transportation_service_provider_id = transportation_service_providers.id 
WHERE tsp_users.email = 'leo_spaceman_tsp_1@example.com';
```

Copy an shipment id and go to the swagger endpoint to try it out here: http://tsplocal:3000/api/v1/docs#/service_agents/indexServiceAgents

## Code Review Verification Steps

* [x] Code follows the guidelines for [Logging](https://github.com/transcom/mymove/blob/master/docs/backend.md#logging)
* [x] The requirements listed in
 [Querying the Database Safely](https://github.com/transcom/mymove/blob/master/docs/backend.md#querying-the-database-safely)
 have been satisfied.
* [x] Request review from a member of a different team.
* [x] Have the Pivotal acceptance criteria been met for this change?

## References

* [Pivotal story](https://www.pivotaltracker.com/story/show/159751456) for this change

### CODE REVIEW

File: pkg/models/service_agent.go [hunk 1]
```diff
@@ -76,6 +76,20 @@ func (s *ServiceAgent) ValidateUpdate(tx *pop.Connection) (*validate.Errors, err
...
+func FetchServiceAgentsByTSP(tx *pop.Connection, tspID uuid.UUID, shipmentID uuid.UUID) ([]ServiceAgent, error) {
+
+	serviceAgents := []ServiceAgent{}
+
+	err := tx.
+		Where("shipments.id = $1 AND shipment_offers.transportation_service_provider_id = $2", shipmentID, tspID).
+		LeftJoin("shipments", "service_agents.shipment_id=shipments.id").
+		LeftJoin("shipment_offers", "shipments.id=shipment_offers.shipment_id").
+		All(&serviceAgents)
+
```
Comment (kilbergr) [inline pkg/models/service_agent.go]: Please include 
`if err != nil {
		return nil, err
	}`

File: swagger/api.yaml [hunk 2]
```diff
@@ -573,9 +573,10 @@ definitions:
         format: uuid
         example: c56a4180-65aa-42ec-a945-5fd21dec0538
         readOnly: true
-      shipment:
-        $ref: '#/definitions/Shipment'
-        x-nullable: true
+      shipment_id:
```
Comment (kilbergr) [inline swagger/api.yaml]: I remember why it was weird to have this--cause then it's in the query params and in the body both.

Comment (kilbergr) [reply swagger/api.yaml]: Anyway, if you remove the example and make it readonly, that's fine.

Comment (chrisgilmerproj) [reply swagger/api.yaml]: What happens if I remove the example? I see other readOnly fields with examples.  Does it just make it less confusing in the swagger docs? Or does something show up differently in the payload?

**Pull request merged**